### PR TITLE
chore(mergify): enhance mergify

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
package updates are not automated and we have to look through even minor ones. minor updates should be approved + merged automatically

## Solution
update dependabot config to merge minor packages automatically. we auto-merge snyk cos is security issue (open to also changing this to minor)

### CODE REVIEW

File: .github/mergify.yml [hunk 1]
```diff
@@ -3,9 +3,8 @@ pull_request_rules:
     conditions:
       - author~=^dependabot\[bot\]$
       - check-success~=lint
-      - check-success~=build
       - check-success~=test
-      - title~=Bump [^\s]+ from ([\d]+)\..+ to \1\.
+      - dependabot-update-type = version-update:semver-minor
```
Comment (kishore03109) [inline .github/mergify.yml:7]: what about patches?

File: .github/mergify.yml [hunk 2]
```diff
@@ -16,9 +15,9 @@ pull_request_rules:
     conditions:
       - author=snyk-bot
       - check-success~=lint
-      - check-success~=build
```
Comment (kishore03109) [inline .github/mergify.yml:19]: may i get context for this regarding why this line was removed?

File: .github/mergify.yml (hunk 1 above)
Comment (seaerchin) [reply .github/mergify.yml]: updated to auto-update for all non-major

File: .github/mergify.yml (hunk 2 above)
Comment (seaerchin) [reply .github/mergify.yml:19]: added back in, removed by accident.

File: .github/mergify.yml [hunk 3]
```diff
@@ -3,8 +3,9 @@ pull_request_rules:
     conditions:
       - author~=^dependabot\[bot\]$
       - check-success~=lint
+      - check-success~=build
       - check-success~=test
-      - dependabot-update-type = version-update:semver-minor
+      - -dependabot-update-type = version-update:semver-major
```
Comment (kishore03109) [inline .github/mergify.yml:8]: wait am a bit confused, should this not be minor then?

Comment (seaerchin) [reply .github/mergify.yml:8]: Screenshot 2024-04-09 at 1 35 38 PM

didn't duplicate the comment here but this negates the condition so this is essentially everything but major (minor + patches)

**Review by kishore03109 — APPROVED** [review]

**Pull request merged**